osu.Game/Overlays/Settings/Sections/DebugSettings/MemorySettings.cs

```diff
@@ -24,73 +26,110 @@ private void load(GameHost host, RealmAccess realm)
             SettingsButton blockAction;
             SettingsButton unblockAction;
 
-            Children = new Drawable[]
+            Add(new SettingsButton
             {
-                new SettingsButton
+                Text = @"Clear all caches",
+                Action = () =>
                 {
-                    Text = @"Clear all caches",
-                    Action = host.Collect
-                },
-                new SettingsButton
+                    host.Collect();
+
+                    // host.Collect() uses GCCollectionMode.Optimized, but we should be as aggressive as possible here.
+                    GC.Collect(GC.MaxGeneration, GCCollectionMode.Aggressive, true, true);
+                }
+            });
+
+            SettingsEnumDropdown<GCLatencyMode> latencyModeDropdown;
+            Add(latencyModeDropdown = new SettingsEnumDropdown<GCLatencyMode>
+            {
+                LabelText = "GC mode",
+            });
+
+            latencyModeDropdown.Current.BindValueChanged(mode =>
+            {
+                switch (mode.NewValue)
                 {
-                    Text = @"Compact realm",
-                    Action = () =>
+                    case GCLatencyMode.Default:
+                        // https://github.com/ppy/osu-framework/blob/1d5301018dfed1a28702be56e1d53c4835b199f2/osu.Framework/Platform/GameHost.cs#L703
+                        GCSettings.LatencyMode = System.Runtime.GCLatencyMode.SustainedLowLatency;
```

Comment:
Note that you'll need to make sure to tell users to not change this during gameplay, because it will be clobbered by `HighPerformanceSession`.

Reply:
From what I read of `HighPerformanceSession`, I gathered that it will handle this fine. It only resets if it's still in the `LowLatency` state, otherwise it's a no-op.